crates/sui-types/src/transaction.rs

```diff
@@ -2978,10 +2978,18 @@ pub enum InputObjectKind {
 
 impl InputObjectKind {
     pub fn object_id(&self) -> ObjectID {
+        self.full_object_id().id()
+    }
+
+    pub fn full_object_id(&self) -> FullObjectID {
         match self {
-            Self::MovePackage(id) => *id,
-            Self::ImmOrOwnedMoveObject((id, _, _)) => *id,
-            Self::SharedMoveObject { id, .. } => *id,
+            Self::MovePackage(id) => FullObjectID::Fastpath(*id),
+            Self::ImmOrOwnedMoveObject((id, _, _)) => FullObjectID::Fastpath(*id),
```

Comment:
nit: i think it would be better to call it `QualifiedObjectID` instead of `FullObjectID`, because this stuff isn't really part of the ID itself, its a function of how we are referring to it

Reply:
discussed on chat, but response for posterity

The prefix Full* as a term used to refer to the complete object identifier inclusive of both the address and the start_version for consensus objects was introduced a couple months ago already in https://github.com/MystenLabs/sui/pull/20822

Fair enough if you want to reopen the discussion on that naming choice, but it would be a broad rename across like 20 files that is not really related to this particular PR